src/Evolution/Systems/NewtonianEuler/FiniteDifference/AoWeno.hpp

```diff
@@ -0,0 +1,173 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <boost/functional/hash.hpp>
+#include <cstddef>
+#include <limits>
+#include <memory>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/FixedHashMap.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/Structure/MaxNumberOfNeighbors.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/DgSubcell/Tags/NeighborData.hpp"
+#include "Evolution/Systems/NewtonianEuler/FiniteDifference/Reconstructor.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+
+namespace NewtonianEuler::fd {
+/*!
+ * \brief Adaptive-order WENO reconstruction hybridizing orders 5 and 3. See
+ * ::fd::reconstruction::aoweno_53() for details.
+ */
+template <size_t Dim>
+class AoWeno53Prim : public Reconstructor<Dim> {
+ private:
+  // Conservative vars tags
+  using MassDensityCons = Tags::MassDensityCons;
+  using EnergyDensity = Tags::EnergyDensity;
+  using MomentumDensity = Tags::MomentumDensity<Dim>;
+
+  // Primitive vars tags
+  using MassDensity = Tags::MassDensity<DataVector>;
+  using Velocity = Tags::Velocity<DataVector, Dim>;
+  using SpecificInternalEnergy = Tags::SpecificInternalEnergy<DataVector>;
+  using Pressure = Tags::Pressure<DataVector>;
+
+  using prims_tags =
+      tmpl::list<MassDensity, Velocity, SpecificInternalEnergy, Pressure>;
+  using cons_tags = tmpl::list<MassDensityCons, MomentumDensity, EnergyDensity>;
+  using flux_tags = db::wrap_tags_in<::Tags::Flux, cons_tags, tmpl::size_t<Dim>,
+                                     Frame::Inertial>;
+  using prim_tags_for_reconstruction =
+      tmpl::list<MassDensity, Velocity, Pressure>;
+
+ public:
+  struct GammaHi {
+    using type = double;
+    static constexpr Options::String help = {
+        "The linear weight for the 5th-order stencil."};
+  };
+  struct GammaLo {
```

Comment:
[not well-formulated enough to be a change request] I don't looooove (GammaHi, GammaLo), I think I'd prefer spelling it out (GammaHigh, GammaLow) or maybe using the numerical values (Gamma5, Gamma3)? I don't know. I don't love these suggestions either...

Reply:
agreed, the reason I went with `Hi` and `Lo` is because that's what the paper does :shrug:

Reply:
That's good enough for me 👍